include/pipert/NetworkPacket.h

```diff
@@ -0,0 +1,49 @@
+#ifndef _NETWORKPACKET_H_
+#define _NETWORKPACKET_H_
+
+#include <type_traits>
+
+#include "pipert/Packet.h"
+#include "pipert/Timer.h"
+
+namespace pipert {
+
+/// Represents data which is transmitted between computers for processing.
+///
+/// NetworkPacket is a special type of Packet, which is used for pipeline
+/// commnunication between separated computers on the network.
+/// See Packet class for details.
+///
+/// \tparam T Type of user data which this packet will contain.
+///           Type must be Trivially Copyable to be appropriate for
+///           network communication.
+template <class T>
+class NetworkPacket : public Packet<T> {
+ public:
+  /// Default constructor for NetworkPacket.
+  NetworkPacket() = default;
+  /// Constructs a NetwrokPacket with the given timestamp and data.
+  /// Type of data is required to be Trivially Copyable type and it
+  /// is being checked during compile time.
+  /// \param timestamp The exact time when this data or the
+  ///                  observed phenomenon was recorded.
+  ///                  See PacketBase::timestamp() for details.
+  /// \param data Predefined data with type T. It comes from
+  ///             Packet::data().
+  NetworkPacket(Timer::Time timestamp, T& data);
+  ~NetworkPacket() = default;
+
+  NetworkPacket(const NetworkPacket<T>&) = default;
+  NetworkPacket& operator=(const NetworkPacket&) = default;
+};
+
+template <class T>
+NetworkPacket<T>::NetworkPacket(Timer::Time timestamp, T& data)
+    : Packet<T>(timestamp, data) {
+  static_assert(std::is_trivially_copyable<T>::value,
```

Comment:
This trait checking and the possibility to create an empty NetworkPacket (it's necessary when receiving packets) are the new things here. Now the default constructors of Packet and PacketBase are protected, and these are used only from the NetworkPacket. I can make it public but is creating empty Packets a valid use case in a local pipeline?
About the [TriviallyCopyable](https://en.cppreference.com/w/cpp/named_req/TriviallyCopyable) type, wouldn't it be a bit strict requirement in a local pipeline? As I see, e.g. there are some test cases where we use at least one complex structure.

Reply:
Ok, leave it this way, especially because the constructor is a totally valid distinction. However, I think that the test which has complex stuff in a packet is a bad example. It would make sense to create an issue on that, so the test should be changed to reflect suggested usage instead of a bad one.